include/NovelRT.Interop/Ecs/NrtEcsTypedefs.h

```diff
@@ -0,0 +1,41 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_INTEROP_ECS_ECSTYPEDEFS_H
+#define NOVELRT_INTEROP_ECS_ECSTYPEDEFS_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    typedef struct NrtCatalogue* NrtCatalogueHandle;
+    typedef struct NrtComponentBufferMemoryContainer* NrtComponentBufferMemoryContainerHandle;
+    typedef struct NrtComponentBufferMemoryContainer_ImmutableDataView*
+        NrtComponentBufferMemoryContainer_ImmutableDataViewHandle;
+    typedef struct NrtComponentCache* NrtComponentCacheHandle;
+    typedef struct NrtEntityIdVector* NrtEntityIdVectorHandle;
+    typedef struct NrtEntityCache* NrtEntityCacheHandle;
+    typedef struct NrtSparseSetMemoryContainer* NrtSparseSetMemoryContainerHandle;
+    typedef struct NrtSparseSetMemoryContainer_ByteIteratorView* NrtSparseSetMemoryContainer_ByteIteratorViewHandle;
+    typedef struct NrtSparseSetMemoryContainer_ConstByteIteratorView*
+        NrtSparseSetMemoryContainer_ConstByteIteratorViewHandle;
+    typedef struct NrtSparseSetMemoryContainer_Iterator* NrtSparseSetMemoryContainer_IteratorHandle;
+    typedef struct NrtSparseSetMemoryContainer_ConstIterator* NrtSparseSetMemoryContainer_ConstIteratorHandle;
+    typedef struct NrtSystemScheduler* NrtSystemSchedulerHandle;
+    typedef struct NrtUnsafeComponentView* NrtUnsafeComponentViewHandle;
+
+    typedef void (*NrtComponentUpdateFnPtr)(NrtSparseSetMemoryContainer_ByteIteratorViewHandle,
+                                            NrtSparseSetMemoryContainer_ByteIteratorViewHandle,
+                                            size_t);
+
+    typedef void (*NrtSystemUpdateFnPtr)(NrtTimestamp, NrtCatalogueHandle);
+
+    typedef NrtAtom NrtEntityId;
+    typedef NrtAtom NrtComponentTypeId;
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif // NOVELRT_INTEROP_ECS_ECSTYPEDEFS_H
```

Comment:
How come these have all been extracted from their relevant headers?

Reply:
Because there was an inconsistency overall across the headers.

This allows all typedefs to be defined up front and simplifies the dependency graph between files. It allows someone working with bindings to see all typedefs/opaque handles in one location so they can easily create the relevant renames/remappings.

(Many libraries, including Vulkan, Xlib, DirectX, etc centralize their own forward declarations at the top of the central include file as well).